core/tx_multicurrency_priceheap_test.go

```diff
@@ -320,7 +370,52 @@ func TestIsCheaper(t *testing.T) {
 	var notCheaper CurrencyCmpFn = func(p1 *big.Int, c1 *common.Address, p2 *big.Int, c2 *common.Address) int {
 		return 1
 	}
-	assert.True(t, cheaper.IsCheaper(tx1, tx2))
-	assert.False(t, equal.IsCheaper(tx1, tx2))
-	assert.False(t, notCheaper.IsCheaper(tx1, tx2))
+	assert.True(t, cheaper.Cmp(tx1, tx2, nil, nil) <= 0)
+	assert.True(t, equal.Cmp(tx1, tx2, nil, nil) <= 0)
+	assert.False(t, notCheaper.Cmp(tx1, tx2, nil, nil) <= 0)
+}
+
+// TestMulticurrencyUnderpriced tests that the underpriced method from pricedList functions
+// properly when handling many different currencies.
+func TestMulticurrencyUnderpriced(t *testing.T) {
```

Comment:
Could you reorganize this test to be a table test with different initial state and different cases for `Underpriced` calls? Now the state of all heaps is mixed up so it's very hard to follow the logic of the final assertions. Then we would have more meaningful test cases like:
- Empty pool, any TX is not underpriced.
- Single native TX in the pool, a cheaper incoming TX is underpriced.
- Two native TXs in the pool, an incoming TX must not be cheaper than the cheapest one.
- 1 TX in native pool, 1 TX in another pool, an incoming TX must not be cheaper than the cheapest of the two.

Reply:
@hbandura I added a more structured approach in https://github.com/celo-org/celo-blockchain/commit/3e7ae1c835261e969f4059e40d59cccced8b97fc , feel free cherry-pick that commit if you like the apporach.

Reply:
@palango thanks, I'll use that

Reply:
I added the remaining cases Valentin mentioned.